### ISSUE TITLE
Implementing a wait cursor for the duration of an event handler is extremely difficult

### ISSUE THREAD

**Issue by rqsanderson** [dotnet/docs#7898]
This is an area that has been problematic for what seems like more than two decades now.  It has always been very hit or miss to successfully change the mouse pointer at runtime.  DoEvents is a workaround, but can introduce dangerous side effects.  Many sources suggest using a separate thread, or otherwise refactoring the long process into a separate event, but that requires a ridiculous amount of effort and extra complexity for a robust line of business application with 1000s or more event handlers that could use a wait cursor.  It's 2018 already, wish this could be addressed.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a43bba3b-c9e6-217f-9781-d1283819b0a3
* Version Independent ID: cf9e74de-2d52-0f07-91b5-788d0d0a840d
* Content: [Mouse Pointers in Windows Forms](https://docs.microsoft.com/en-us/dotnet/framework/winforms/mouse-pointers-in-windows-forms)
* Content Source: [docs/framework/winforms/mouse-pointers-in-windows-forms.md](https://github.com/dotnet/docs/blob/master/docs/framework/winforms/mouse-pointers-in-windows-forms.md)
* Product: **dotnet-framework**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

**Comment by Thraka** [comment]
@rqsanderson Hi. This seems more like product feedback and not really related to a problem with that documentation page. Please correct me if I'm wrong. 

If I understand you correctly you seem to have problems changing the cursor to something else (like the wait cursor for example) and trying to get things done in the background, but something else comes through and changes your cursor on you. This being a side effect of using the `DoEvents` call. And you've read advice that you shouldn't use `DoEvents` and instead should look into moving that code into other threads/events. Yes?

This seems like just a general issue with Windows Forms per the discussion [here on StackOverflow](https://stackoverflow.com/questions/7517271/c-sharp-preventing-clicks-on-a-form-without-using-enabled-false) Does WPF solve that? I don't know, it is a UI framework designed completely different from Windows Forms. I'm not suggesting you upgrade, I'm just pointing out that I think you're just running into a design limitation you'll need to work around, which sucks. 😞

**Comment by rqsanderson** [comment]
Thank you for reaching out, I appreciate it.

Actually, I am pleased to report we have been able to overcome the difficulties in our scenarios.  It turns there were dialog boxes shown during some of our events, which had the effect of canceling the wait cursor.  We found that setting the wait cursor again after the dialog fixed part of the problem.  Additionally, we found some existing stray calls to DoEvents.  In this case, DoEvents was also canceling the wait cursor.  Removing the calls to DoEvents fixed that problem as well, so I think we are in good shape for the time being.  Perhaps the situation with the cursor has improved more than I thought.
